client/src/document/index.scss

```diff
@@ -35,6 +35,57 @@ a.new {
 .main-page-content {
   padding: math.div($base-spacing, 2) $base-spacing $base-spacing;
 
+  .code-example {
+    position: relative;
+
+    .copy-icon {
+      position: absolute;
+      background: #eee url("../assets/clippy.svg") 50% 50% no-repeat;
+      background-position: 61% 45%;
+      background-size: 17px 17px;
+      vertical-align: middle;
+      border: 1px solid lightgray;
+      top: math.div($base-spacing, 3);
+      right: math.div($base-spacing, 3);
+      width: 33px;
+      height: 36px;
+      opacity: 0.4;
+      margin: 0;
+      padding: 7px math.div($base-spacing, 4);
+      border-radius: 5px;
+      z-index: 5;
+    }
+
+    .user-message {
+      position: absolute;
+      top: 15px;
+      right: math.div($base-spacing, 3);
+      display: none;
+      background: #292929;
+      padding: 0.55em 0.5em;
+      border-radius: math.div($base-spacing, 6);
+      color: #fff;
+      opacity: 1;
+      z-index: 5;
+      font: 16px/1.25 arial, sans-serif;
+      font-size: 14px;
+    }
+
+    .show {
+      display: block;
+    }
+  }
+
+  .code-example:focus .copy-icon,
+  .code-example:hover .copy-icon {
+    opacity: 1;
+  }
+
+  .copy-icon:hover {
+    background-color: rgb(225, 225, 225);
+    cursor: pointer;
+  }
+
```

Comment:
This works great! Thank you, @tannerdolby - Did a bit of mostly refactoring

```suggestion
.code-example {
    position: relative;

    .copy-icon {
      background: $neutral-550 url("../assets/clippy.svg") center center
        no-repeat;
      background-position: 61% 45%;
      background-size: 16px;
      border: 1px solid $neutral-550;
      border-radius: $default-border-radius;
      cursor: pointer;
      height: 36px;
      margin: 0;
      padding: math.div($base-spacing, 4);
      position: absolute;
      right: math.div($base-spacing, 3);
      top: math.div($base-spacing, 3);
      vertical-align: middle;
      width: 33px;
      z-index: $bottom-layer;

      @media #{$mq-large-desktop-and-up} {
        opacity: 0.4;

        &:hover,
        &:focus {
          background-color: $neutral-525;
          opacity: 1;
        }
      }
    }

    .user-message {
      background: $neutral-100;
      border-radius: $default-border-radius;
      color: $neutral-600;
      display: none;
      font-size: $small-font-size;
      opacity: 1;
      padding: math.div($base-spacing, 4);
      position: absolute;
      right: math.div($base-spacing, 3);
      top: 15px;
      z-index: $bottom-layer;
    }

    &:focus,
    &:hover {
      .copy-icon {
        opacity: 1;
      }
    }

    .show {
      display: block;
    }
  }
```

Reply:
@schalkneethling Did you mean to remove the `opacity: 0.4` from the initial state of `.copy-icon`? I like everything you submitted but was curious if we wanted it to be completely visible before a user hovered or focused the `.code-example` container.